Objects/typeobject.c

```diff
@@ -5779,7 +5792,17 @@ void
 _PyType_SetFlags(PyTypeObject *self, unsigned long mask, unsigned long flags)
 {
     BEGIN_TYPE_LOCK();
-    type_set_flags_with_mask(self, mask, flags);
+    unsigned long new_flags = (self->tp_flags & ~mask) | flags;
+    if (new_flags != self->tp_flags) {
+#ifdef TYPE_FLAGS_UPDATE_NEEDS_STOP
+        PyInterpreterState *interp = _PyInterpreterState_GET();
```

Comment:
if the world is stopped when assigning flags then I think type lock above would be unnecessary

Reply:
Fixed as suggested.